### ISSUE TITLE
ICE in nativeMembers() when trying to initialize a constant struct using a hex literal

### ISSUE THREAD

**Issue by agroce** [ethereum/solidity#9951]
## Description

This contract:
```solidity
contract C {
    struct S {
        mapping(uint => uint) c;
    }
    S public constant e = 0x1212121212121212121212121212121212121212;
}
```

produces:

```
Internal compiler error during compilation:
/home/user/solidity/libsolidity/ast/Types.cpp(2403): Throw in function virtual MemberList::MemberMap solidity::frontend::StructType::nativeMembers(const solidity::frontend::ASTNode *) const
Dynamic exception type: boost::exception_detail::clone_impl<solidity::langutil::InternalCompilerError>
std::exception::what: 
[solidity::util::tag_comment*] = 
```

when compiled with solc.

On master, using AFL fuzzing.  Another discovery using https://github.com/agroce/afl-compiler-fuzzer.

## Environment

- Compiler version:  0.7.3-develop.2020.10.2+commit.756e21a8.Linux.clang
- Target EVM version (as per compiler settings): N/A
- Framework/IDE (e.g. Truffle or Remix): N/A
- EVM execution environment / backend / blockchain client: N/A
- Operating system: Ubuntu 18.04 in docker

## Steps to Reproduce

Above shows pretty clearly, I think.

**Comment by cameel** [comment]
This started producing an ICE in 0.7.0. Constant structs are not yet implemented AFAIK (#4037) and even then, they can't be initialized with a hex literal but it should be a normal compiler error instead.

### Output on 0.6.12
```solidity
Error: Type address payable is not implicitly convertible to expected type struct C.S memory.
 --> /tmp/test.sol:5:27:
  |
5 |     S public constant e = 0x1212121212121212121212121212121212121212;
  |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Constants of non-value type not yet implemented.
 --> /tmp/test.sol:5:5:
  |
5 |     S public constant e = 0x1212121212121212121212121212121212121212;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Internal or recursive type is not allowed for public state variables.
 --> /tmp/test.sol:5:5:
  |
5 |     S public constant e = 0x1212121212121212121212121212121212121212;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by axic** [comment]
This is a still happening on 0.7.x (and 0.8.x). It would be nice to fix before 0.8.0.
